shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/metadata/schema/builder/SchemaBuilder.java

```diff
@@ -64,42 +63,61 @@
     static {
         ShardingSphereServiceLoader.register(DialectTableMetaDataLoader.class);
     }
-    
+
     /**
-     * Build ShardingSphere schema.
-     * 
+     * build actual and logic table meta data.
+     *
      * @param materials schema builder materials
-     * @return ShardingSphere schema
+     * @return actual and logic table meta data
      * @throws SQLException SQL exception
      */
-    public static ShardingSphereSchema build(final SchemaBuilderMaterials materials) throws SQLException {
-        ShardingSphereSchema result = new ShardingSphereSchema();
-        addRuleConfiguredTables(materials, result);
-        appendRemainTables(materials, result);
+    public static Map<Map<String, TableMetaData>, Map<String, TableMetaData>> build(final SchemaBuilderMaterials materials) throws SQLException {
```

Comment:
`Map<Map<String, TableMetaData>, Map<String, TableMetaData>>` seems a little wired, maybe we need a table Name field in `TableMetaData`? How do you think?

Reply:
Yes, I think that makes more sense, but I've noticed that a lot of places use maps, so I've gone that route. If adding a TableName property is a bit too much to change, can I fix it in the next PR?

Reply:
Sure.

Reply:
ok ~